### PR TITLE
Refactor Accordion.js to use hooks

### PR DESCRIPTION
**Pull request by husseyexplores**
#### What does this PR do?
This PR refactors Accordion.js to use hooks.

#### Where should the reviewer start?
`/src/js/components/Accordion/Accordion.js`

#### What testing has been done on this PR?
Storybook

#### How should this be manually tested?
Storybook

#### Any background context you want to provide?
Nothing, but I do have a question. Why are we using `getDerivedStateFromProps` here? What is the use case? I don't see any difference in storybook examples even after removing that method. I implemented the hooks version of it anyway though. The results still make no difference with or without it while testing it manually in storybook examples.

#### What are the relevant issues?
#3394 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Comment by husseyexplores** [comment]
Some Accordion snapshot tests are failing (were already failing before I started working on it) because of the swapping CSS classes.
For example, snapshot contains this: ` className="c10 c1"` , but the render in the test contains this: `className="c1 c10"`. I'm not sure what to do regarding this, update the snapshots or not.

File: src/js/components/Accordion/Accordion.js [hunk 1]
```diff
@@ -7,85 +7,76 @@ import { AccordionContext } from './AccordionContext';
...
-    const { animate, children, messages, ...rest } = this.props;
-    const { activeIndexes } = this.state;
-
-    delete rest.onActive;
-
-    return (
-      <Box role="tablist" {...rest}>
-        {Children.toArray(children).map((panel, index) => (
-          <AccordionContext.Provider
-            key={`accordion-panel_${index + 0}`}
```
Comment (husseyexplores) [inline src/js/components/Accordion/Accordion.js]: One more question here, does `index + 0` won't make any difference I believe? Should I remove that?

Comment (husseyexplores) [reply src/js/components/Accordion/Accordion.js]: Oh I get it now. It is there to silence `eslint(react/no-array-index-key)` error message.

File: src/js/components/Accordion/Accordion.js [hunk 2]
```diff
@@ -7,85 +7,76 @@ import { AccordionContext } from './AccordionContext';
...
+          }}
+        >
+          {panel}
+        </AccordionContext.Provider>
+      ))}
+    </Box>
+  );
 }
 
+Accordion.defaultProps = {
```
Comment (ShimiSun) [inline src/js/components/Accordion/Accordion.js]: Can we embed the defaultProps values as part of the props declaration on lines #10-#18

Comment (husseyexplores) [reply src/js/components/Accordion/Accordion.js]: I thought it was not the recommended way of doing it (idk why though) and just came to know it is indeed the preferred way. Thanks for letting me know. I just made the change and also rearranged the props alphabetically.

**Comment by ShimiSun** [comment]
Thank you for your contribution! 🎉

**Pull request merged**